ipinfo/cmd_download.go

```diff
@@ -238,3 +268,38 @@ func unzipWrite(file *os.File, data io.Reader) error {
 
 	return nil
 }
+
+func computeSHA256(filepath string) (string, error) {
+	file, err := os.Open(filepath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	hasher := sha256.New()
+	if _, err := io.Copy(hasher, file); err != nil {
```

Comment:
Is this a streaming hasher or one big memory load and then hash? Is it possible to stream the memory into the hash function so we don't have to load the whole thing at once?

Reply:
`io.Copy` streams data into the hash function rather than loading the whole file at once. It's processed by chunks of 32KB

[Stackoverflow. It has got a few comments stating this](https://stackoverflow.com/questions/15879136/how-to-calculate-sha256-file-checksum-in-go)
Also asked gpt

Reply:
Nice